core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java

```diff
@@ -429,7 +429,7 @@ public void testTasksListWaitForTimeout() throws Exception {
                 .execute();
         try {
             // Wait for the task to start on all nodes
-            assertBusy(() -> assertEquals(internalCluster().numDataAndMasterNodes(), client().admin().cluster().prepareListTasks()
+            assertBusy(() -> assertEquals(internalCluster().size(), client().admin().cluster().prepareListTasks()
```

Comment:
The problem was in a mismatch in how the `TransportTestTaskAction` decided where to send the requests. It used to send it only to data nodes and the test here used `numDataAndMasterNodes` to check for things. This only works when you have master/data nodes. When you introduce dedicated masters, it's broken. I opted to fix this by sending the tasks to all nodes, not caring if they are data/master or coordinating. I felt this is more inline with our new way of making coord nodes (previously known as client nodes) as first class citizens.